### PR TITLE
[bufffer] Got rid of x() and App type-hint (and more)

### PR DESCRIPTION
**Pull request by Quix0r**
Cleanups/fixes:
- purged deprecated x() and used !empty() instead
- add App as type-hint where missing
- added curly braces and spaces for better readability/code convention
- open curly brace after method/function should be next line

### CODE REVIEW

File: buffer/buffer.php [hunk 1]
```diff
@@ -16,50 +16,56 @@
...
+	Addon::unregisterHook('notifier_normal',         'addon/buffer/buffer.php', 'buffer_send');
+	Addon::unregisterHook('jot_networks',            'addon/buffer/buffer.php', 'buffer_jot_nets');
 	Addon::unregisterHook('connector_settings',      'addon/buffer/buffer.php', 'buffer_settings');
 	Addon::unregisterHook('connector_settings_post', 'addon/buffer/buffer.php', 'buffer_settings_post');
 }
 
-function buffer_module() {}
-
-function buffer_content(&$a) {
+function buffer_module()
```
Comment (tobiasd) [inline buffer/buffer.php]: For those cases, I would really appreciate the one-liner, even though the braces might have to go to the next lines according to the standard.

Comment (MrPetovan) [reply buffer/buffer.php]: I've never liked this way of signalling the addon provides a module. I feel like we should check for the existence of either `addon_content()` or `addon_post()`.

Comment (Quix0r) [reply buffer/buffer.php]: And continue rewriting it to a object-oriented way. Like with filter chains?

Comment (MrPetovan) [reply buffer/buffer.php]: Ideally, addons would provide a `Module` class and hook into the routing mechanism to add their custom routes. Unfortunately at the moment we only have the former, with an implicit router that assumes that all modules are in `src/Module`.

**Pull request merged**